### ISSUE TITLE
[css-masking-1] Clarify behaviour of clipPathUnits="objectBoundingBox" when placed on a child clipPath

### ISSUE THREAD

**Issue by BigBadaboom** [w3c/fxtf-drafts#247]
Section 6.1 states wrt clipPathUnits:

> The coordinate system has its origin at the top left corner of the bounding box of the element to which the clipping path applies to...

But what "element" is the correct one when your clipPath itself has a clipPath?

```
<svg width="3in" height="3in" viewBox="0 0 300 300" version="1.1"
     xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">

  <defs>
    <clipPath id="Clip1" clip-path="url(#Clip2)">
      <rect width="200" height="200"/>
    </clipPath>
    <clipPath id="Clip2" clipPathUnits="objectBoundingBox">
      <circle cx="0.5" cy="0.5" r="0.5"/>
    </clipPath>
  </defs>

  <rect fill="none" stroke="blue"
        x="1" y="1" width="298" height="298"/>

  <rect fill="red" x="0" y="0" width="300" height="300" clip-path="url(#Clip1)"
        transform="translate(50,50)"/>

</svg>
```

https://jsfiddle.net/xqs5rmda/

Here, the child `<clipPath>` with `id="Clip2"` has `objectBoundingBox` specified. But are we referring to the bounds of the parent clipPath, or the original `<rect>` element?  I would have assumed the former, but browsers are not consistent:

- Chrome: rect
- Firefox: rect
- IE!: clipPath
- Batik: result bugged, but is using clipPath

Behaviour of Chrome and Firefox seems like a bug to me.  But either way, the language should probably be expanded to make it clear what the correct behaviour should be.

**Comment by dirkschulze** [comment]
I agree that the specification text can be interpreted to use the bounding box of the `<clipPath>` element. However, the `<clipPath>` element itself doesn't have a bounding box. (Related to another issue #28 and `getBBox`.) So this would need to get specified and implemented.

The same question applies to `<mask>`.

IMO the behavior should be consistent to the behavior of `<pattern>` and `patternUnits` where a pattern may reference another pattern. Here the bounding box of the graphics element gets used for all pattern references. This would be consistent to the behavior that is implemented in:
* Safari, Chrome and Firefox

**Comment by AmeliaBR** [comment]
Related issue raised on Firefox because of interop failures: https://bugzilla.mozilla.org/show_bug.cgi?id=1531890

That issue is about `clip-path` property applied to a *child* of a `<clipPath>` element, and I'm confident calling that a Firefox bug — but I agree that the spec could be clarified.

For this issue (`clip-path` property on the `<clipPath>` element itself), I agree with Dirk that it is less clear what *should* happen.  Both Chromium and Firefox use the original element's user space for the secondary clipping path ([test case CodePen](https://codepen.io/AmeliaBR/collab/fd314d690be3c48a72b238d695db395e)), and that is consistent with the spec describing this structure as an intersection of the two clipping paths.  But it still needs to be made explicit.

cc @longsonr
